public/app/plugins/datasource/cloud-monitoring/components/ConfigEditor/JWTConfig.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FormEvent, useState } from 'react';
+import { startCase, isObject } from 'lodash';
+import { Button, FileUpload, InlineField, Input, useStyles, Alert } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme } from '@grafana/data';
+
+const configKeys = [
+  'type',
+  'project_id',
+  'private_key_id',
+  'private_key',
+  'client_email',
+  'client_id',
+  'auth_uri',
+  'token_uri',
+  'auth_provider_x509_cert_url',
+  'client_x509_cert_url',
+];
+
+export interface JWT {
+  token_uri: string;
+  client_email: string;
+  private_key: string;
+  project_id: string;
+}
+
+export interface Props {
+  onChange: (jwt: JWT) => void;
+  isConfigured: boolean;
+}
+
+const validateJson = (json: { [key: string]: string }) => isObject(json) && configKeys.every((key) => !!json[key]);
```

Comment:
if you make this something like `(json): json is JWT` you shouldn't need the `json as JWT` type assertion below (but really, you're still doing the same cast but just with slightly cleaner semantics).

If you wanted to be more safe you could make the argument `unknown` and then pick each value out and return a new object of the values (or throw/return undefined if its not valid).

Reply:
neat! thanks for feedback